File: lib/pacer/pipe/label_prefix_pipe.rb
module Pacer::Pipes
  class LabelPrefixPipe < RubyPipe
    def initialize(prefix)
      super()
      @prefix = /^#{prefix}/
    end

    def processNextStart
      while true
        edge = @starts.next
        return edge if edge.label =~ @prefix
      end
    end
  end
end

File: ext/src/main/java/pacer/pipe/LabelPrefixPipe.java
package com.xnlogic.pacer.pipes;

import com.tinkerpop.blueprints.Edge;
import com.tinkerpop.pipes.AbstractPipe;
import java.util.regex.Pattern;

public class LabelPrefixPipe extends AbstractPipe<Edge, Edge> {
    private Pattern pattern;
  
    public LabelPrefixPipe(final String pattern) {
        super();
        this.pattern = Pattern.compile("^" + pattern);
    }

    protected Edge processNextStart() {
        while (true) {
            Edge e = this.starts.next();
            if (this.pattern.matcher(e.getLabel()).matches()) {
                return e;
            }
        }
    }
}
